plugins/BEdita/API/tests/TestCase/Identifier/OAuth2IdentifierTest.php

```diff
@@ -153,4 +153,33 @@ public function testIdentify(?array $expected, array $credentials, array $oauthR
         }
         static::assertSame($expected, $result);
     }
+
+    /**
+     * Test `identify` method with callback.
+     *
+     * @return void
+     * @covers ::identify()
+     */
+    public function testIdentifyWithCallback(): void
+    {
+        $authProvider = $this->fetchTable('AuthProviders')->get(1);
+        $authProvider->set('params', [
+            'options' => [
+                'credentials_callback' => function (array $credentials): bool {
+                    return $credentials['provider_username'] === 'first_user';
+                },
+            ],
+        ]);
+
+        $identifier = new OAuth2Identifier();
+        $identifier->setConfig(compact('authProvider'));
+
+        $result = $identifier->identify([
+            'auth_provider' => 'example',
+            'provider_username' => 'first_user',
+            'access_token' => 'very-log-string',
+        ]);
+
+        static::assertNotEmpty($result);
```

Comment:
This looks fine. We should provide another test for `return false callback`, and `assertNull`

Reply:
Done